modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -3,18 +3,47 @@ package types
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	yaml "gopkg.in/yaml.v2"
 
 	crypto "github.com/cosmos/cosmos-sdk/crypto/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	connectiontypes "github.com/cosmos/ibc-go/modules/core/03-connection/types"
 )
 
 const (
-	IcaPrefix string = "ics27-1-"
+	ICAPrefix string = "ics-27"
```

Comment:
the spec specifies two different formats:
`ics27-1` and `ics-27`

my preference is not to reference any version number in the ports, we should never have a reason to change the port ID generation as backwards compatibility/migration would be a pain, it'd be a different module at that point

Either `ics27` or `ics-27` sounds fine to me

Reply:
Lets go with `ics-27` :)

Reply:
I actually just remembered why I kept the `-1` version. I figured in the situation whereby we might have a version `-2` we may still have interchain accounts from version `1` still active and we will want to differentiate. A version two could potentially have 1 owner for N accounts rather than 1 to 1 mapping.

Reply:
Can't this be represented in the actual interchain accounts version?

Reply:
Let's say we have a bunch of interchain accounts registered with version 1. If we upgrade the architecture to version 2 we have to either 1. keep supporting the version 1 account or 2 provide a migration path for all of the currently existing interchain accounts. I imagined the first choice to be easier and considered using the version in the port-id as a future-proofing step if it's ever required. 

It's a minimal addition that may save time in the future. It probably will never come up, but if it does it could be useful.

Reply:
But version 1 portID's still need to be supported? You cannot change port ID's (as this is negotiated in the handshake)

Reply:
I meant that the functionality associated with an account registered with version 1 of interchain accounts would still need to be supported and by keeping the version number in the prefix here would be a quick way to distinguish between what accounts were registered with what version of the module.

Reply:
If a DAO registers an account on the hub with version 1 of interchain accounts, but version 2 (hypothetically) ended up having a different architecture, we would need to either migrate this account (ownership + any tokens) to version a version 2 account or keep supporting version 1.

Reply:
The channel would still be accessible on both the host and controller side even if versions changed? My main hesitation is it seems odd to overload the port ID with information that could be contained in places it is designated for (app version in the channel version)

Reply:
Ah, you're saying if both sides update to version 2 of interchain accounts, it will be possible to get the version string that was originally used to create the channel, and that we can use that to distinguish between accounts registered under different versions?

Getting very edge case here, but what would happen if the channel closed? Would we still be able to access the version string?

Reply:
It depends on the structure of tracking the active channels. The channel version is always accessible even if the channel is closed. The active channel mapping would just need to know the previous channel which was active

Reply:
Alright, let's go with this for now. Before we go live with ICA I think it's a good idea to have some plan in place for how we would handle this kind of issue if it arises.